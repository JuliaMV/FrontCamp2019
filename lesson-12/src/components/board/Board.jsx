/* eslint-disable camelcase */
import React from 'react';
import PropTypes from 'prop-types';

import Card from 'components/card/Card';

import css from './Board.module.css';

class Board extends React.Component {
  getCards = ({
    title, release_date, poster_path, genres, id,
  }) => (
    <Card
      title={title}
      date={release_date}
      img={poster_path}
      genres={genres}
      id={id}
      key={id}
    />
  );

  noFilms = () => (<li className={css.noFilms}>No films found</li>);

  render() {
    const { films } = this.props;
    const content = films.length > 0 ? films.map(this.getCards)
      : this.noFilms();

    return (
      <ul className={css.container}>
        { content }
      </ul>
    );
  }
}

Board.propTypes = {
  films: PropTypes.arrayOf(PropTypes.object).isRequired,
};

export default Board;
