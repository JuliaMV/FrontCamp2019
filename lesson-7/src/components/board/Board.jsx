/* eslint-disable camelcase */
import React from 'react';

import Card from 'components/card/Card';
import responce from '../../data/movies';

import css from './Board.module.css';

class Board extends React.Component {
  state = {
    films: [],
  };

  componentDidMount() {
    this.setState({
      films: responce.data.slice(),
      // films: []
    });
  }

  getCards = ({
    title, release_date, poster_path, genres, id,
  }) => (<Card title={title} date={release_date} img={poster_path} genres={genres} id={id} />);

  noFilms = () => (<li className={css.noFilms}>No films found</li>);

  render() {
    const { films } = this.state;
    return (
      <ul className={css.container}>
        {films.length > 0 ? films.slice(0, 6).map(this.getCards) : this.noFilms()}
      </ul>
    );
  }
}


export default Board;
