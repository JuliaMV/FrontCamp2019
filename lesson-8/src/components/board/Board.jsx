/* eslint-disable camelcase */
import React from 'react';

// import constants from 'src/constants';

import Card from 'components/card/Card';
import responce from '../../data/movies';


import css from './Board.module.css';

class Board extends React.Component {
  // state = {
  //   films: [],
  // };

  // componentDidMount() {
  //   this.setState({
  //     films: responce.data.slice(),
  //     // films: []
  //   });
  // }

  getCards = ({
    title, release_date, poster_path, genres, id,
  }) => (
    <Card
      title={title}
      date={release_date}
      img={poster_path}
      genres={genres}
      id={id}
      key={id}
    />
  );

  noFilms = () => (<li className={css.noFilms}>No films found</li>);

  render() {
    const { films } = this.props
    const content = films.length > 0 ? films.map(this.getCards)
      : this.noFilms();

    return (
      <ul className={css.container}>
        { content }
      </ul>
    );
  }
}


export default Board;
