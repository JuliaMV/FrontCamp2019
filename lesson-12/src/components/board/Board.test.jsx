import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';

import Board from './Board';

Enzyme.configure({ adapter: new Adapter() });


describe('Board component', () => {
  test('should render without films', () => {
    const props = {
      films: [],
    };
    // eslint-disable-next-line react/jsx-props-no-spreading
    const component = shallow(<Board {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  test('should render with 1 film', () => {
    const props = {
      films: [
        {
          img: 'http://image.com',
          title: 'News title',
          genres: ['fantasy'],
          date: '03.02.2019',
          id: 1325,
        },
      ],
    };
    // eslint-disable-next-line react/jsx-props-no-spreading
    const component = shallow(<Board {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  test('should render with 3 films', () => {
    const props = {
      films: [
        {
          img: 'http://image.com',
          title: 'News title',
          genres: ['fantasy'],
          date: '03.02.2019',
          id: 1325,
        },
        {
          img: 'http://image.com',
          title: 'Film title',
          genres: ['action'],
          date: '03.02.2020',
          id: 132500,
        },
        {
          img: 'http://image.com',
          title: 'Lalala',
          genres: ['comedy', 'drama'],
          date: '03.03.2019',
          id: 135,
        },
      ],
    };
    // eslint-disable-next-line react/jsx-props-no-spreading
    const component = shallow(<Board {...props} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
