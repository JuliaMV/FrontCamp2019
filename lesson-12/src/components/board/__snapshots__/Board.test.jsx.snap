// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board component should render with 1 film 1`] = `
<ul>
  <withRouter(Card)
    genres={
      Array [
        "fantasy",
      ]
    }
    id={1325}
    key="1325"
    title="News title"
  />
</ul>
`;

exports[`Board component should render with 3 films 1`] = `
<ul>
  <withRouter(Card)
    genres={
      Array [
        "fantasy",
      ]
    }
    id={1325}
    key="1325"
    title="News title"
  />
  <withRouter(Card)
    genres={
      Array [
        "action",
      ]
    }
    id={132500}
    key="132500"
    title="Film title"
  />
  <withRouter(Card)
    genres={
      Array [
        "comedy",
        "drama",
      ]
    }
    id={135}
    key="135"
    title="Lalala"
  />
</ul>
`;

exports[`Board component should render without films 1`] = `
<ul>
  <li>
    No films found
  </li>
</ul>
`;
